fiftyone/utils/hf_hub.py

```diff
@@ -0,0 +1,527 @@
+        _add_detections_field(fo_dataset, hf_dataset, feature_name)
+    elif isinstance(feature_type, datasets.Sequence):
+        subfeature = feature_type.feature
+        fo_subtype = _convert_hf_value_dtype(subfeature)
+        fo_dataset.add_sample_field(
+            feature_name, fo.ListField, subfield=fo_subtype
+        )
+        vals = hf_dataset[feature_name]
+        fo_dataset.set_values(feature_name, vals)
+
+
+def _convert_hf_dataset_to_fiftyone(
+    hf_dataset, repo_id, image_field=None, **kwargs
+):
+    if "name" not in kwargs:
+        kwargs["name"] = repo_id
+    print("Converting dataset")
+    fo_dataset = fo.Dataset(**kwargs)
+    download_dir = _get_download_dir(repo_id)
+    if image_field is None:
+        image_field = _get_image_field_name(hf_dataset)
+
+    def _add_split_to_fo_dataset(
+        main_fo_dataset, hf_split, image_field, download_dir, split_name=None
+    ):
+        tmp_fo_dataset = fo.Dataset()
+        _add_images(
+            tmp_fo_dataset,
+            hf_split,
+            image_field,
+            download_dir,
+            split_name=split_name,
+        )
+        tmp_fo_dataset.compute_metadata()
+        for feature_name in hf_split.features:
+            if feature_name == image_field:
+                continue
+            _convert_hf_type_to_fiftyone(
+                tmp_fo_dataset, hf_split, feature_name
+            )
+        main_fo_dataset.merge_samples(tmp_fo_dataset)
+
+    if isinstance(hf_dataset, datasets.DatasetDict):
+        split_names = list(hf_dataset.keys())
+        for split_name in split_names:
+            print(f"Processing split '{split_name}'")
+            split = hf_dataset[split_name]
+            _add_split_to_fo_dataset(
+                fo_dataset,
+                split,
+                image_field,
+                download_dir,
+                split_name=split_name,
+            )
+    else:
+        _add_images(fo_dataset, hf_dataset, image_field, download_dir)
+        fo_dataset.compute_metadata()
+        _add_split_to_fo_dataset(
+            fo_dataset,
+            hf_dataset,
+            image_field,
+            download_dir,
+        )
+
+    return fo_dataset
+
+
+class BaseHuggingFaceLoader:
+    def __init__(self, repo_id: str, hf_dataset, **kwargs):
+        self.repo_id = repo_id
+        self.hf_dataset = hf_dataset
+        self.kwargs = kwargs
+
+    def load(self):
+        raise NotImplementedError
+
+
+class DefaultHuggingFaceLoader(BaseHuggingFaceLoader):
+    def __init__(self, repo_id: str, hf_dataset, **kwargs):
+        super().__init__(repo_id, hf_dataset, **kwargs)
+
+    def load(self):
+        return _convert_hf_dataset_to_fiftyone(
+            self.hf_dataset, self.repo_id, **self.kwargs
+        )
+
+
+def _is_convertable_to_fiftyone(repo_id):
+    api = hfh.HfApi()
+    return api.file_exists(repo_id, "fiftyone.py", repo_type="dataset")
+
+
+def _load_fiftyone_dataset_from_hub(repo_id, **kwargs):
```

Comment:
Does this also work with parquet files? an increasing number of datasets are created using datasets and `push_to_hub` so will be stored as Parquet.